### DISCUSSION TITLE
I want to use Next JS for e-commerce platform how do I create product page that can be crawl and have good SEO?

### DISCUSSION THREAD

**Discussion by shawwal** [vercel/next.js#15313] [Help]
I have an existing API to display product listing and product page. What is the good practice to make the pages index by Google Crawl properly?

**Comment by leerob** [comment]
As long as your page is "pre-rendered", meaning the HTML generated will come from the server, then it will be easily crawled by the Google bot. You can either use Next's static site generation methods (`getStaticProps`/`getStaticPaths`) or using server-side rendering (`getServerSideProps`).

**Reply by mdathersajjad** [reply]
What would you suggest for a project which needs to generate atleast 10k or 100k pages or more. Generating at deployment will cause delay in deployment. Using fallback : true in getStaticPaths and only generating a handful of pages at deployment will leave the other pages to be not indexed properly by google because the pages will be generated on the fly the first time its requested. Is my assumption correct @leerob ? Even if it was indexed previously when it reindexes it will see only the html skeleton right ?

**Reply by mdathersajjad** [reply]
If the build and cached files are not completely wiped during deployment it will speed up the deployment processes right ?
